controller/space_iterations_test.go

```diff
@@ -308,31 +325,39 @@ func (rest *TestSpaceIterationREST) TestWICountsWithIterationListBySpace() {
 	// when
 	_, cs := test.ListSpaceIterationsOK(rest.T(), svc.Context, svc, ctrl, fxt.Spaces[0].ID, nil, nil)
 	// then
-	require.Len(rest.T(), cs.Data, 5)
-	for _, iterationItem := range cs.Data {
-		if uuid.Equal(*iterationItem.ID, fxt.Iterations[1].ID) {
-			assert.Equal(rest.T(), 5, iterationItem.Relationships.Workitems.Meta[KeyTotalWorkItems])
-			assert.Equal(rest.T(), 2, iterationItem.Relationships.Workitems.Meta[KeyClosedWorkItems])
-		} else if uuid.Equal(*iterationItem.ID, fxt.Iterations[2].ID) {
-			// we expect these counts should include that of child iterations too.
-			expectedTotal := 0 + 4 + 5  // sum of all items of self + child + grand-child
-			expectedClosed := 0 + 0 + 5 // sum of closed items self + child + grand-child
-			assert.Equal(rest.T(), expectedTotal, iterationItem.Relationships.Workitems.Meta[KeyTotalWorkItems])
-			assert.Equal(rest.T(), expectedClosed, iterationItem.Relationships.Workitems.Meta[KeyClosedWorkItems])
-		} else if uuid.Equal(*iterationItem.ID, fxt.Iterations[3].ID) {
-			// we expect these counts should include that of child iterations too.
-			expectedTotal := 4 + 5  // sum of all items of self and child
-			expectedClosed := 0 + 5 // sum of closed items of self and child
-			assert.Equal(rest.T(), expectedTotal, iterationItem.Relationships.Workitems.Meta[KeyTotalWorkItems])
-			assert.Equal(rest.T(), expectedClosed, iterationItem.Relationships.Workitems.Meta[KeyClosedWorkItems])
-		} else if uuid.Equal(*iterationItem.ID, fxt.Iterations[4].ID) {
-			// we expect these counts should include that of child iterations too.
-			expectedTotal := 5 + 0  // sum of all items of self and child
-			expectedClosed := 5 + 0 // sum of closed items of self and child
-			assert.Equal(rest.T(), expectedTotal, iterationItem.Relationships.Workitems.Meta[KeyTotalWorkItems])
-			assert.Equal(rest.T(), expectedClosed, iterationItem.Relationships.Workitems.Meta[KeyClosedWorkItems])
-		}
+	require.Len(rest.T(), cs.Data, len(fxt.Iterations))
+	expectedTotalCounts := map[string]int{
```

Comment:
Should we mention that these counts are for type-group = Iterations?
just a comment or in the variable name would help.
At least for me, it took some time to figure it out :smiley:

Reply:
It did for me as well, so yes.

Reply:
@pranavgore09 done here: 6ae59cbcc7b35213aa623d1258e02b0654b30bc8